modules/migrations/gitea_downloader.go

```diff
@@ -683,3 +679,20 @@ func (g *GiteaDownloader) GetReviews(index int64) ([]*base.Review, error) {
 	}
 	return allReviews, nil
 }
+
+// fixPullHeadSha is a workaround for https://github.com/go-gitea/gitea/issues/12675
+// When no head sha is available, this is because the branch got deleted in the base repo.
+// pr.Head.Ref points in this case not to the head repo branch name, but the base repo ref,
+// which stays available to resolve the commit sha.
+func fixPullHeadSha(client *gitea_sdk.Client, pr *gitea_sdk.PullRequest) error {
+	owner := pr.Base.Repository.Owner.UserName
+	repo := pr.Base.Repository.Name
+	if pr.Head != nil && pr.Head.Sha == "" {
+		headCommit, _, err := client.GetSingleCommit(owner, repo, url.PathEscape(pr.Head.Ref))
```

Comment:
`GetSingleCommit` didn't work for me in tea (https://gitea.com/gitea/tea/pulls/340), I had to use `GetRepoRefs()` to make it work..?

Reply:
Get single commut worjed quite well for me, would be nice @noerw uf you can tell me what you have tryed, ... getRefs should be fine too ...

Reply:
I have tested some things ... **GetSingleCommit** works *mostly* but **GetRepoRefs** works more reliable :D

will switch to it